### DISCUSSION TITLE
Orthotropic elasticity fill method showing error

### DISCUSSION THREAD

**Discussion by chakra34** [idaholab/moose#25325] [Q&A General] [answered]
Hi,

I am using the following elasticity tensor for orthotropic material:

  [elasticity_tensor]
    type = ComputeElasticityTensor
    fill_method = orthotropic
    C_ijkl = '4.95e8 4.95e8 1.78e9 9.549e6 2.8568e8 2.8568e8 0.06925 0.249717818 0.249717818 0.06925 0.069444 0.069444'
    block = 'block1'
  []

And upon executing I get the following error:
*** ERROR ***
Orthotropic elasticity tensor input is not consistent with symmetry requirements. Check input for accuracy

When I look at the source of the error (https://mooseframework.inl.gov/docs/doxygen/moose/classRankFourTensorTempl.html#a0ffa73b46895112e75bc64265041569f) in the doxygen, I checked the conditions and it seems to me everything looks consistent between the Young's modulus and Poisson's ratio.

Is the error because of precision or am I missing something completely ?

Thanks,
Aritra

**Comment by GiudGiud** [comment]
Your input is failing this
```
815   // Input must satisfy constraints.
  816   bool preserve_symmetry = [MooseUtils::absoluteFuzzyEqual(nuab * Eb, nuba * Ea) &&
  817                            [MooseUtils::absoluteFuzzyEqual(nuca * Ea, nuac * Ec) &&
  818                            [MooseUtils::absoluteFuzzyEquall(nubc * Ec, nucb * Eb);
  819 
  820   if (!preserve_symmetry)
  821     [mooseError](https://mooseframework.inl.gov/docs/doxygen/moose/MooseError_8h.html#aa959168eed29caa7aaa85ffc1319a9ad)("Orthotropic elasticity tensor input is not consistent with symmetry requirements. "
  822                "Check input for accuracy");
```

**Reply by GiudGiud** [reply]
so yeah it could be that you are not inputting enough digits to satisfy these equalities

**Reply by chakra34** [reply]
I thought so.. I was actually inputting all the digits in a regular calculator, lol. But I will think of something

**Reply by chakra34** [reply]
@GiudGiud yeah it was the precision issue, and the following works:

  [elasticity_tensor]
    type = ComputeElasticityTensor
    fill_method = orthotropic
    C_ijkl = '5.0e8  5.0e8  2.0e9  9.549e6 2.8568e8 2.8568e8    0.06925   0.277776   0.277776  0.06925   0.069444     0.069444'
    block = 'skin_region core'
  []



Another question, for Compute Thermal Expansion EigenStrain 

![image](https://github.com/idaholab/moose/assets/22983992/5bfdbd38-9525-4c06-a618-cd0b6bdeea01)


Can we use an alpha tensor ? so different alpha_xx, alpha_yy, and alpha_zz by entering the thermal expansion coefficient as a tensor ? or does it take only scalar values ?

Thanks,
Aritra

**Reply by GiudGiud** [reply, marked as answer]
only takes a scalar here
feel free to modify
